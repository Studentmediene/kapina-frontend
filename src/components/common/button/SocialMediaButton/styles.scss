@import '../../../../variables.scss';

.socialMediaButton {
  width: $social-media-button-width;
  margin-left: $social-media-button-margin;
  margin-right: $social-media-button-margin;
  margin-bottom: $social-media-button-margin + 5px;
}

.link {
  &:focus {
    // Gray, since the buttons can be over light or dark background
    outline-color: #888;
    // Also add some brightness, since the outline is more subtle
    filter: brightness(115%);
  }
}
