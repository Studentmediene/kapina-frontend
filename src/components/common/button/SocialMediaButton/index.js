import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.scss';

const SocialMediaButton = props => (
  <a target="_blank" rel="noopener noreferrer" href={props.link}>
    <img
      src={props.image}
      alt={props.text}
      className={styles.socialMediaButton}
    />
  </a>
);

SocialMediaButton.propTypes = {
  link: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default SocialMediaButton;
