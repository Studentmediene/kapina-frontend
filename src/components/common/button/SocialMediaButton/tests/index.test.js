import React from 'react';
import { shallow } from 'enzyme';

import SocialMediaButton from '../';

describe('<SocialMediaButton />', () => {
  it('renders correctly', () => {
    const props = {
      link: 'link',
      image: 'image',
      text: 'text',
    };
    const tree = shallow(<SocialMediaButton {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
