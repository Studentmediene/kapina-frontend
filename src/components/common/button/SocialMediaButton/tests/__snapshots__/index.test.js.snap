// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialMediaButton /> renders correctly 1`] = `
<a
  href="link"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="text"
    className="socialMediaButton"
    src="image"
  />
</a>
`;
